addons/stock/models/stock_move_line.py

```diff
@@ -249,18 +249,19 @@ def _apply_putaway_strategy(self):
         if self._context.get('avoid_putaway_rules'):
             return
         self = self.with_context(do_not_unreserve=True)
-        for package, smls in groupby(self, lambda sml: sml.result_package_id):
+        for (location, package), smls in groupby(self, lambda sml: (sml.location_dest_id, sml.result_package_id)):
             smls = self.env['stock.move.line'].concat(*smls)
+            locations = smls[0].location_dest_id.child_internal_location_ids
```

Comment:
```suggestion
            locations = location.child_internal_location_ids
```
Isn't it the same given the `groupby`?